pkg/backend/httpstate/console.go

```diff
@@ -16,23 +16,46 @@ package httpstate
 
 import (
 	"net/url"
+	"os"
 	"path"
 	"strings"
 )
 
+const (
+	// PulumiCloudURL is the Cloud URL used if no environment or explicit cloud is chosen.
+	PulumiCloudURL = "https://" + defaultAPIDomainPrefix + "pulumi.com"
+
+	// defaultAPIDomainPrefix is the assumed Cloud URL prefix for typical Pulumi Cloud API endpoints.
+	defaultAPIDomainPrefix = "api."
+	// defaultConsoleDomainPrefix is the assumed Cloud URL prefix typically used for the Pulumi Console.
+	defaultConsoleDomainPrefix = "app."
+)
+
+// cloudConsoleURL returns a URL to the Pulumi Cloud Console, rooted at cloudURL. If there is
+// an error, returns "".
 func cloudConsoleURL(cloudURL string, paths ...string) string {
 	u, err := url.Parse(cloudURL)
 	if err != nil {
 		return ""
 	}
 
 	switch {
+	case os.Getenv("PULUMI_CONSOLE_DOMAIN") != "":
```

Comment:
I debated it, and went with hard-coding it because it was only used in one place. But I'll make the change, since it's probably a better idea to export it.